tests/unittests/core/fileio/fileutilstest.cpp

```diff
@@ -0,0 +1,313 @@
+          << "\nActual:\n" << resultStr
+          << "\nExpected:\n" << expStr;
+  }
+};
+
+/*******************************************************************************
+ *  Test Methods
+ ******************************************************************************/
+
+TEST_F(FileUtilsTest, testReadExistingFile) {
+  auto p = FileUtils::readFile(rootFile);
+
+  EXPECT_EQ("test\n", p.toStdString());
+}
+
+TEST_F(FileUtilsTest, testReadNonexistentFileShouldThrow) {
+  EXPECT_THROW(FileUtils::readFile(rootFileMissing), Exception);
+}
+
+TEST_F(FileUtilsTest, testWrittenDataShouldBeReadBack) {
+  FileUtils::writeFile(rootFile, "someData\n");
+  auto p = FileUtils::readFile(rootFile);
+
+  EXPECT_EQ("someData\n", p.toStdString());
+}
+
+TEST_F(FileUtilsTest, testCopyValidFile) {
+  FileUtils::copyFile(rootFile, rootFileCopy);
+  auto p1 = FileUtils::readFile(rootFile);
+  auto p2 = FileUtils::readFile(rootFileCopy);  // throws if file not found
+
+  EXPECT_EQ("test\n", p1.toStdString());
+  EXPECT_EQ("test\n", p2.toStdString());
+}
+
+TEST_F(FileUtilsTest, testCopyNonexistingFileShouldThrow) {
+  EXPECT_THROW(FileUtils::copyFile(rootFileMissing, rootFileCopy), Exception);
+}
+
+TEST_F(FileUtilsTest, testMoveValidFile) {
+  FileUtils::move(rootFile, rootFileCopy);
+  auto p = FileUtils::readFile(rootFileCopy);  // throws if file not found
+
+  EXPECT_THROW(FileUtils::readFile(rootFile), Exception);
+  EXPECT_EQ("test\n", p.toStdString());
+}
+
+TEST_F(FileUtilsTest, testMoveNonexistingFileShouldThrow) {
+  EXPECT_THROW(FileUtils::move(rootFileMissing, rootFileCopy), Exception);
+}
+
+TEST_F(FileUtilsTest, testRemoveValidFile) {
+  FileUtils::removeFile(rootFile);
+
+  EXPECT_THROW(FileUtils::readFile(rootFile), Exception);
+}
+
+TEST_F(FileUtilsTest, testRemoveNonexistingFileShouldThrow) {
+  EXPECT_THROW(FileUtils::removeFile(rootFileMissing), Exception);
+}
+
+TEST_F(FileUtilsTest, testCreateSubdir) {
+  FileUtils::makePath(subdirCopy);
+
+  EXPECT_TRUE(subdirCopy.isExistingDir());
+}
+
+TEST_F(FileUtilsTest, testRecursiveRemoveSubdir) {
+  FileUtils::removeDirRecursively(subdir);
+
+  EXPECT_FALSE(subdir.isExistingDir());
+  EXPECT_FALSE(subdirFile.isExistingFile());
+}
+
+TEST_F(FileUtilsTest, testRecursiveCopySubdir) {
+  FileUtils::copyDirRecursively(subdir, subdirCopy);
+
+  // ensure source remain unchanged
+  EXPECT_TRUE(subdir.isExistingDir());
+  EXPECT_TRUE(subdirFile.isExistingFile());
+  EXPECT_TRUE(subdirSubdirFile.isExistingFile());
+  EXPECT_TRUE(subdirSubdirFileHidden.isExistingFile());
+
+  // ensure destination is complete copy
+  EXPECT_TRUE(subdirCopy.isExistingDir());
+  EXPECT_TRUE(subdirCopyFile.isExistingFile());
+  EXPECT_TRUE(subdirCopySubdirFile.isExistingFile());
+  EXPECT_TRUE(subdirCopySubdirFileHidden.isExistingFile());
+}
+
+TEST_F(FileUtilsTest, testFindDirectories) {
+  auto p = FileUtils::findDirectories(root);
+
+  ExpectList(p, {subdir});
+}
+
+TEST_F(FileUtilsTest, testFindFileInDirectory) {
```

Comment:
Just a note (can be left as-is): Generally try to keep the test names consistent with the function under test, it makes reading tests and their output more easy and intuitive. For example here the function under test is called `getFilesInDirectory` so the test could be called `testGetFilesInDirectory` ("get" vs. "find", "file" vs. "files"). This way it's immediately clear from the test name which function it tests.

If there are multiple tests for the same function, of course additional terms need to be added to the test name - ideally (but not necessarily) at the end of the test name.

Reply:
Yes, you have right there, my original intent was to describe what the test does initially (the obvious result) when making the test instead of the function name that is little bit cryptic for me. 

And I forgot to rename it your way before opening PR. I'll correct it.